framework/src/media/voice/SpeechDetector.cpp

```diff
@@ -0,0 +1,104 @@
+/******************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <debug.h>
+#include <functional>
+
+#include "SpeechDetector.h"
+#include "SoftwareKeywordDetector.h"
+#include "SoftwareEndPointDetector.h"
+#include "HardwareKeywordDetector.h"
+#include "HardwareEndPointDetector.h"
+
+namespace media {
+namespace voice {
+
+static SpeechDetector *g_instance = nullptr;
+
+SpeechDetectorInterface *SpeechDetectorInterface::instance()
+{
+	if (!g_instance) {
+		g_instance = new SpeechDetector();
```

Comment:
To implement singleton, this way can be problem.
How about using `static SpeechDector instance`?

Reply:
You're right. I'll fix this!